dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -2491,9 +2495,8 @@ public void refreshFeatures(Filter filter) {
         refreshFeatures();
     }
 
-    boolean missingRequiredJava(FeatureResource fr) {
-        Integer requiredJava = fr.getRequireJava();
-        return requiredJava == null ? false : JavaInfo.majorVersion() < requiredJava;
+    boolean withinJavaRange(FeatureResource fr) {
+        return range == null ? true :  fr.getJavaRange().includes(JAVA_MAJOR_VERSION);
```

Comment:
This won't compile now.  range is undefined after you update.
```suggestion
        VersionRange range = fr.getJavaRange();
        return range == null ? true : range.includes(JAVA_MAJOR_VERSION);
 ``